packages/core/js-client/src/compilerSupport/registerService.ts

```diff
@@ -28,27 +28,22 @@ interface RegisterServiceArgs {
   service: ServiceImpl;
 }
 
+// This function iterates on plain object or class instance functions ignoring inherited functions and prototype chain.
 const findAllPossibleRegisteredServiceFunctions = (
   service: ServiceImpl,
-): Set<string> => {
-  let prototype: Record<string, unknown> = service;
-  const serviceMethods = new Set<string>();
+): Array<string> => {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  const prototype = Object.getPrototypeOf(service) as ServiceImpl;
```

Comment:
maybe move this directly inside if statement because it seems to be needed only if `isClassInstance`